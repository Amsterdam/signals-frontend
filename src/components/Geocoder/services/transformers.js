export const wktPointToPointFeature = wktPoint => {
  if (!wktPoint.includes('POINT')) {
    throw TypeError('Provided WKT geometry is not a point.');
  }
  const coordinate = wktPoint.split('(')[1].split(')')[0];
  const lat = parseFloat(coordinate.split(' ')[1]);
  const lng = parseFloat(coordinate.split(' ')[0]);

  return {
    lat,
    lng,
  };
};
