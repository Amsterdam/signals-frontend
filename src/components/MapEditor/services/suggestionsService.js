const GEOCODER_API_SUGGEST =
  'https://geodata.nationaalgeoregister.nl/locatieserver/v3/suggest?fq=gemeentenaam:(amsterdam OR weesp)&fq=type:adres&q=';

const getSuggestions = async searchTerm => {
  const result = await window.fetch(`${GEOCODER_API_SUGGEST}${searchTerm}`);
  const data = await result.json();
  const suggestions = data.response.docs.map(item => ({
    id: item.id,
    name: item.weergavenaam,
  }));
  return suggestions;
};

export default getSuggestions;
