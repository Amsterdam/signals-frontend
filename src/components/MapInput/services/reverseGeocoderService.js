import proj4 from 'proj4';

proj4.defs(
  'EPSG:28992',
  `+proj=sterea +lat_0=52.15616055555555 +lon_0=5.38763888888889 +k=0.9999079 +x_0=155000 +y_0=463000 +ellps=bessel +towgs84=565.417,50.3319,465.552,-0.398957,0.343988,-1.8774,4.0725 +units=m +no_defs`
);

export const transformCoords = proj4('EPSG:4326', 'EPSG:28992');
export const serviceURL =
    'http://geodata.nationaalgeoregister.nl/locatieserver/revgeo?type=adres&rows=1';

function formatRequest(baseUrl, xy, distance = 50) {
  const xyRD = transformCoords.forward(xy);
  return `${baseUrl}&X=${xyRD.x}&Y=${xyRD.y}&distance=${distance}`;
}

const reverseGeocoderService = async location => {
  const xy = {
    x: location.lng,
    y: location.lat,
  };
  const url = formatRequest(serviceURL, xy);
  const result = await fetch(url);
  const data =  await result.json();
  return data?.response?.docs[0]?.weergavenaam || '';
};

export default reverseGeocoderService;
