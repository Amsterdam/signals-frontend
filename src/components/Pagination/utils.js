export const FILLER = 'FILLER';
export const NEXT = 'NEXT';
export const PREVIOUS = 'PREVIOUS';

/**
 * Gets the previous valid index
 *
 * @param   {number} currentPage - current page index
 * @returns {number} previous page index
 */
export const getPreviousIndex = currentPage =>
  currentPage === 1 ? currentPage : currentPage - 1;

/**
 * Gets the next valid index
 *
 * @param   {number} currentPage - current page index
 * @param   {number} totalPages - total number of pages
 * @returns {number} next page index
 */
export const getNextIndex = (currentPage, totalPages) =>
  currentPage === totalPages ? currentPage : currentPage + 1;

/**
 * Get an array of 1-based numbers
 *
 * @param {number} from - First number in the array
 * @param {number} to - Last number in the array
 * @returns {number[]}
 */
export const getRange = (from, to) =>
  [...new Array(to).keys()].map(key => key + 1).filter(key => key >= from);

/**
 * Get an array of page numbers, filler indicators and navigation indicators
 *
 * @param   {number} currentPage - current page index
 * @param   {number} totalPages - total number of pages
 * @returns {(string|number)[]}
 */
export const pageNumbersList = (currentPage, totalPages) => {
  /**
   * totalNumbers: the total page numbers to show on the control; pages on either side of
   * the current page + the optional spill or navigation buttons
   */
  const totalNumbers = 5;

  if (totalPages <= totalNumbers) {
    return getRange(1, totalPages);
  }

  const startPage = Math.max(2, currentPage - 1);
  const endPage = Math.min(totalPages - 1, currentPage + 1);

  const pages = getRange(startPage, endPage);

  /**
   * hasLeftSpill: has hidden pages to the left
   * hasRightSpill: has hidden pages to the right
   */
  const hasLeftSpill = startPage > 2;
  const hasRightSpill = totalPages - endPage > 1;

  const items = [
    currentPage > 1 && PREVIOUS,
    1,
    hasLeftSpill && FILLER,
    ...pages,
    hasRightSpill && FILLER,
    totalPages,
    currentPage < totalPages && NEXT,
  ];

  return items.filter(Boolean);
};
