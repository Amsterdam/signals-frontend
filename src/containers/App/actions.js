import {
  AUTHENTICATE_USER, SHOW_GLOBAL_ERROR, LOGIN, LOGOUT
} from './constants';

export function authenticateUser(credentials) {
  return {
    type: AUTHENTICATE_USER,
    payload: credentials
  };
}

export function showGlobalError(message) {
  return {
    type: SHOW_GLOBAL_ERROR,
    payload: message
  };
}

export function doLogin(domain) {
  return {
    type: LOGIN,
    payload: domain
  };
}

export function doLogout() {
  return {
    type: LOGOUT,
    payload: null
  };
}

