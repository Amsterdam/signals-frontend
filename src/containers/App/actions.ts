// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2019 - 2022 Gemeente Amsterdam
import type { Action } from 'types'

import {
  AUTHENTICATE_USER,
  AUTHORIZE_USER,
  SHOW_GLOBAL_NOTIFICATION,
  RESET_GLOBAL_NOTIFICATION,
  LOGIN_FAILED,
  LOGIN,
  LOGOUT,
  LOGOUT_FAILED,
  UPLOAD_PROGRESS,
  UPLOAD_SUCCESS,
  UPLOAD_FAILURE,
  SET_SEARCH_QUERY,
  RESET_SEARCH_QUERY,
  GET_SOURCES,
  GET_SOURCES_FAILED,
  GET_SOURCES_SUCCESS,
  POST_MESSAGE,
} from './constants'
import type { GlobalNotification, UserCredentials, User, Source } from './types'

export type LoginFailedAction = Action<typeof LOGIN_FAILED, string>
export type LogoutFailedAction = Action<typeof LOGOUT_FAILED, string>
export type AuthenticateUserAction = Action<
  typeof AUTHENTICATE_USER,
  UserCredentials
>
export type AuthorizeUserAction = Action<typeof AUTHORIZE_USER, User>
export type ShowGlobalNotificationAction = Action<
  typeof SHOW_GLOBAL_NOTIFICATION,
  GlobalNotification
>
export type ResetGlobalNotificationAction = Action<
  typeof RESET_GLOBAL_NOTIFICATION,
  never
>
export type DoLoginAction = Action<typeof LOGIN, null>
export type DoLogoutAction = Action<typeof LOGOUT, null>
export type UploadSuccessAction = Action<typeof UPLOAD_SUCCESS, never>
export type UploadProgressAction = Action<typeof UPLOAD_PROGRESS, number>
export type UploadFailureAction = Action<typeof UPLOAD_FAILURE, never>
export type SetSearchQueryAction = Action<typeof SET_SEARCH_QUERY, string>
export type ResetSearchQueryAction = Action<typeof RESET_SEARCH_QUERY, never>
export type GetSourcesAction = Action<typeof GET_SOURCES, never>
export type GetSourcesFailedAction = Action<typeof GET_SOURCES_FAILED, string>
export type GetSourcesSuccessAction = Action<
  typeof GET_SOURCES_SUCCESS,
  Source[]
>
export type PostMessageAction = Action<typeof POST_MESSAGE, string>

export const loginFailed = (payload: string): LoginFailedAction => ({
  type: LOGIN_FAILED,
  payload,
})

export const logoutFailed = (payload: string): LogoutFailedAction => ({
  type: LOGOUT_FAILED,
  payload,
})

export const authenticateUser = (
  credentials: UserCredentials
): AuthenticateUserAction => ({
  type: AUTHENTICATE_USER,
  payload: credentials,
})

export const authorizeUser = (payload: User): AuthorizeUserAction => ({
  type: AUTHORIZE_USER,
  payload,
})

export const showGlobalNotification = (
  payload: GlobalNotification
): ShowGlobalNotificationAction => ({
  type: SHOW_GLOBAL_NOTIFICATION,
  payload,
})

export const resetGlobalNotification = (): ResetGlobalNotificationAction => ({
  type: RESET_GLOBAL_NOTIFICATION,
})

export const doLogin = (): DoLoginAction => ({
  type: LOGIN,
  payload: null,
})

export const doLogout = (): DoLogoutAction => ({
  type: LOGOUT,
  payload: null,
})

export const uploadProgress = (progress: number): UploadProgressAction => ({
  type: UPLOAD_PROGRESS,
  payload: progress,
})

export const uploadSuccess = (): UploadSuccessAction => ({
  type: UPLOAD_SUCCESS,
})

export const uploadFailure = (): UploadFailureAction => ({
  type: UPLOAD_FAILURE,
})

export const setSearchQuery = (payload: string): SetSearchQueryAction => ({
  type: SET_SEARCH_QUERY,
  payload,
})

export const resetSearchQuery = (): ResetSearchQueryAction => ({
  type: RESET_SEARCH_QUERY,
})

export const getSources = (): GetSourcesAction => ({
  type: GET_SOURCES,
})

export const getSourcesFailed = (payload: string): GetSourcesFailedAction => ({
  type: GET_SOURCES_FAILED,
  payload,
})

export const getSourcesSuccess = (
  payload: Source[]
): GetSourcesSuccessAction => ({
  type: GET_SOURCES_SUCCESS,
  payload,
})

export const postMessage = (message: string): PostMessageAction => ({
  type: POST_MESSAGE,
  payload: message,
})

export type AppActionTypes =
  | LoginFailedAction
  | LogoutFailedAction
  | AuthenticateUserAction
  | AuthorizeUserAction
  | ShowGlobalNotificationAction
  | ResetGlobalNotificationAction
  | DoLogoutAction
  | UploadSuccessAction
  | UploadProgressAction
  | UploadFailureAction
  | SetSearchQueryAction
  | ResetSearchQueryAction
  | GetSourcesAction
  | GetSourcesFailedAction
  | GetSourcesSuccessAction
