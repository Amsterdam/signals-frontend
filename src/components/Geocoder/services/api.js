import { wktPointToPointFeature } from './transformers';
import { getLocation } from '../../../shared/services/map-location';

const GEOCODER_API_SUGGEST =
  'https://geodata.nationaalgeoregister.nl/locatieserver/v3/suggest?fq=gemeentenaam:(amsterdam OR weesp)&fq=type:adres&q=';
const GEOCODER_API_LOOKUP =
  'https://geodata.nationaalgeoregister.nl/locatieserver/v3/lookup?fq=gemeentenaam:(amsterdam OR weesp)&fq=type:adres&id=';

export const getSuggestions = async searchTerm => {
  const result = await window.fetch(`${GEOCODER_API_SUGGEST}${searchTerm}`);
  const data = await result.json();
  const suggestions = data.response.docs.map(item => ({
    id: item.id,
    name: item.weergavenaam,
  }));
  return suggestions;
};

export const getAddressById = async addressId => {
  const result = await window.fetch(`${GEOCODER_API_LOOKUP}${addressId}`);
  const { response } = await result.json();
  if (response.docs[0]) {
    const { centroide_ll } = response.docs[0];
    return {
      location: wktPointToPointFeature(centroide_ll),
      address: { ...getLocation(response.docs[0]) },
    };
  }
  return null;
};
