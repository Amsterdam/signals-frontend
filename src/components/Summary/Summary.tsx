// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2020 - 2023 Gemeente Amsterdam
import { useCallback } from 'react'
import type { FC, KeyboardEvent } from 'react'

import { Link, themeSpacing } from '@amsterdam/asc-ui'
import { Marker } from '@amsterdam/react-maps'
import { useDispatch } from 'react-redux'
import { useLocation } from 'react-router-dom'
import styled from 'styled-components'

import Map from 'components/Map'
import configuration from 'shared/services/configuration/configuration'
import { markerIcon } from 'shared/services/configuration/map-markers'
import MAP_OPTIONS from 'shared/services/configuration/map-options'
import { formatAddress } from 'shared/services/format-address'
import type { SummaryProps } from 'signals/incident/components/form/MapSelectors/Asset/types'
import {
  NEARBY_TYPE,
  selectionIsUndetermined,
} from 'signals/incident/components/form/MapSelectors/constants'
import type {
  FeatureType,
  Item,
} from 'signals/incident/components/form/MapSelectors/types'
import { showMap } from 'signals/incident/containers/IncidentContainer/actions'

import onButtonPress from '../../utils/on-button-press'

const mapWidth = 640
const mapHeight = 180
const iconSize = 40

const Wrapper = styled.div`
  position: relative;
  margin: ${themeSpacing(0, 0, 0, 0)};
`

const StyledLink = styled(Link)`
  text-decoration: underline;
  font-size: 1rem;
  cursor: pointer;
`

const StyledMap = styled(Map)`
  max-width: ${mapWidth}px;
  height: ${mapHeight}px;
  margin: ${themeSpacing(0, 0, 4, 0)};
`

const StyledMarker = styled(Marker)`
  cursor: none;
`

const StyledImg = styled.img`
  margin-right: ${themeSpacing(4)};
  flex-shrink: 0;
`

const LocationDescription = styled.div`
  display: flex;
  align-items: center;
  margin-top: ${themeSpacing(2)};
  margin-bottom: ${themeSpacing(3)};
`

const Address = styled.div`
  margin-bottom: ${themeSpacing(3)};
`

const defaultCenter = {
  lat: configuration.map.options.center[0],
  lng: configuration.map.options.center[1],
}

const getIconSrc = (item: Item, featureTypes: FeatureType[]) => {
  if (!item?.type || selectionIsUndetermined(item)) {
    return undefined
  }

  const featureType = featureTypes.find(
    ({ typeValue }) => typeValue === item.type
  )

  return featureType && featureType.icon.iconUrl
}

const Summary: FC<SummaryProps> = ({
  address,
  coordinates,
  selection,
  featureTypes,
}) => {
  const center = coordinates || defaultCenter

  const options = {
    ...MAP_OPTIONS,
    zoom: configuration.map.optionsSummaryMap.zoom,
    attributionControl: false,
    center,
  }

  const { pathname } = useLocation()
  const atFinalSummary = pathname.includes('summary')

  let summaryAddress = address ? formatAddress(address) : ''
  summaryAddress =
    !summaryAddress && coordinates
      ? 'Locatie is gepind op de kaart'
      : summaryAddress

  const dispatch = useDispatch()

  const onKeyUp = useCallback(
    (event: KeyboardEvent<HTMLAnchorElement>) => {
      onButtonPress(event, () => dispatch(showMap()))
    },
    [dispatch]
  )

  return (
    <Wrapper data-testid="asset-select-summary">
      <StyledMap mapOptions={options}>
        <StyledMarker
          args={[center]}
          options={{ icon: markerIcon, keyboard: false }}
        />
      </StyledMap>

      <Address data-testid="asset-select-summary-address">
        {summaryAddress}
      </Address>

      <div>
        {selection?.map((item) => {
          const { id, type } = item || {}

          const summaryDescription =
            type !== NEARBY_TYPE ? item.label : undefined
          const iconSrc = getIconSrc(item, featureTypes)

          return (
            <LocationDescription key={`${id}-address`}>
              {getIconSrc(item, featureTypes) && (
                <StyledImg
                  data-testid="type-icon"
                  alt={item.label}
                  src={iconSrc}
                  height={iconSize}
                  width={iconSize}
                />
              )}
              <div>
                {selection && (
                  <div data-testid="asset-select-summary-description">
                    {summaryDescription}
                  </div>
                )}
              </div>
            </LocationDescription>
          )
        })}
      </div>
      {!atFinalSummary && (
        <StyledLink
          data-testid="map-edit-button"
          onClick={() => dispatch(showMap())}
          onKeyUp={onKeyUp}
          variant="inline"
          tabIndex={0}
        >
          Wijzigen
        </StyledLink>
      )}
    </Wrapper>
  )
}

export default Summary
