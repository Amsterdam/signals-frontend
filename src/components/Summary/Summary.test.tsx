// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2020 - 2022 Gemeente Amsterdam
import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import * as reactRedux from 'react-redux'

import configuration from 'shared/services/configuration/configuration'
import { formatAddress } from 'shared/services/format-address'
import { contextValue as assetSelectContextValue } from 'signals/incident/components/form/MapSelectors/Asset//__tests__/withAssetSelectContext'
import { AssetSelectProvider } from 'signals/incident/components/form/MapSelectors/Asset/context'
import type { SummaryProps } from 'signals/incident/components/form/MapSelectors/Asset/types'
import type { Item } from 'signals/incident/components/form/MapSelectors/types'
import { showMap } from 'signals/incident/containers/IncidentContainer/actions'
import { history } from 'test/utils'
import { withAppContext } from 'test/utils'
import type { Address } from 'types/address'

import MockInstance = jest.MockInstance
import Summary from './Summary'

jest.mock('shared/services/configuration/configuration')
jest.mock('react-router-dom', () => ({
  __esModule: true,
  ...jest.requireActual('react-router-dom'),
}))

const selection: Item[] = [
  {
    id: 'PL734',
    type: 'plastic',
    description: 'Plastic asset',
    location: {},
    label: 'Plastic container - PL734',
  },
]
const featureType = {
  label: 'Plastic',
  description: 'Plastic asset',
  icon: {
    iconUrl: 'plasticIconUrl',
  },
  idField: 'id_nummer',
  typeField: 'fractie_omschrijving',
  typeValue: 'plastic',
}

export const address = {
  postcode: '1000 AA',
  huisnummer: 100,
  woonplaats: 'Amsterdam',
  openbare_ruimte: 'West',
}

export const summaryProps: SummaryProps = {
  selection,
  featureTypes: [featureType],
  address,
  coordinates: { lat: 0, lng: 0 },
}

export const withContext = (
  Component: JSX.Element,
  context = assetSelectContextValue
) =>
  withAppContext(
    <AssetSelectProvider value={context}>{Component}</AssetSelectProvider>
  )

const dispatch = jest.fn()

describe('signals/incident/components/form/AssetSelect/Summary', () => {
  beforeEach(() => {
    jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => dispatch)
    const dispatchEventSpy: MockInstance<any, any> = jest.spyOn(
      global.document,
      'dispatchEvent'
    )
    dispatch.mockReset()
    dispatchEventSpy.mockReset()
  })

  afterEach(() => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    configuration.__reset()
    jest.resetAllMocks()
  })

  it('renders interactive map correctly', () => {
    render(withContext(<Summary {...summaryProps} />))

    expect(screen.getByTestId('asset-select-summary')).toBeInTheDocument()
    expect(
      screen.getByTestId('asset-select-summary-description')
    ).toBeInTheDocument()
    expect(
      screen.getByTestId('asset-select-summary-address')
    ).toBeInTheDocument()
    expect(screen.getByText(/wijzigen/i)).toBeInTheDocument()
    expect(screen.queryByTestId('type-icon')).toBeInTheDocument()
    expect(screen.queryByTestId('map-base')).toBeInTheDocument()
  })

  it('does not render empty values', () => {
    const propsNoFeatureTypes = {
      ...summaryProps,
      featureTypes: [],
    }
    render(withContext(<Summary {...propsNoFeatureTypes} />))

    const idRe = new RegExp(`${selection[0].id}$`)
    const undefinedRe = new RegExp('undefined')

    expect(screen.getByText(idRe)).toBeInTheDocument()
    expect(screen.queryByTestId('type-icon')).not.toBeInTheDocument()
    expect(screen.queryByText(undefinedRe)).not.toBeInTheDocument()
  })

  it('renders without selection', () => {
    const propsNoSelection = {
      ...summaryProps,
      selection: undefined,
    }
    render(withContext(<Summary {...propsNoSelection} />))

    expect(
      screen.queryByTestId('asset-select-summary-description')
    ).not.toBeInTheDocument()
    expect(screen.getByText(formatAddress(address))).toBeInTheDocument()
  })

  it('should call edit by mouse click', () => {
    render(withContext(<Summary {...summaryProps} />))
    expect(dispatch).not.toHaveBeenCalledWith(showMap())

    const element = screen.getByText(/wijzigen/i)

    expect(dispatch).not.toHaveBeenCalledWith(showMap())

    userEvent.click(element)

    expect(dispatch).toHaveBeenCalledWith(showMap())
  })

  it('should call edit by return key', () => {
    render(withContext(<Summary {...summaryProps} />))
    expect(dispatch).not.toHaveBeenCalledWith(showMap())

    const element = screen.getByText(/wijzigen/i)
    element.focus()

    userEvent.keyboard('a')

    expect(dispatch).not.toHaveBeenCalledWith(showMap())

    userEvent.keyboard('{Enter}')

    expect(dispatch).toHaveBeenCalledWith(showMap())
  })

  it('renders summary address', () => {
    const address = summaryProps.address as Address
    const propsNoAddress = {
      ...summaryProps,
      address: undefined,
    }

    const { rerender } = render(withContext(<Summary {...summaryProps} />))
    expect(
      screen.queryByText('Locatie is gepind op de kaart')
    ).not.toBeInTheDocument()
    expect(screen.getByText(formatAddress(address))).toBeInTheDocument()

    rerender(withContext(<Summary {...propsNoAddress} />))

    expect(
      screen.getByText('Locatie is gepind op de kaart')
    ).toBeInTheDocument()
    expect(screen.queryByText(formatAddress(address))).not.toBeInTheDocument()
  })

  it('renders a Map component with the correct iconSrc prop', () => {
    render(withContext(<Summary {...summaryProps} />))
    expect(
      screen
        .getByTestId('map-base')
        .querySelector(`img[src='/assets/images/icon-select-marker.svg']`)
    ).toBeInTheDocument()
  })

  it("renders the mapEditButton at 'incident/vulaan'", () => {
    render(withContext(<Summary {...summaryProps} />))

    expect(screen.getByTestId('map-edit-button')).toBeInTheDocument()
  })

  it("does not render the mapEditButton at 'incident/summary'", async () => {
    history.push('/incident/summary')

    render(withContext(<Summary {...summaryProps} />))

    await waitFor(() => {
      expect(screen.queryByTestId('map-edit-button')).not.toBeInTheDocument()
    })
  })
})
