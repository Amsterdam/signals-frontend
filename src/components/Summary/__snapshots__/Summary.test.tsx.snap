// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signals/incident/components/form/AssetSelect/Summary renders a MapStatic component with the correct iconSrc prop 1`] = `
<span
  data-testid="mapStatic"
>
  <img
    alt=""
  />
</span>
`;
